src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -50,30 +51,87 @@
- * &#64;SuppressWarnings("foo") enum E { }
- * &#64;SuppressWarnings("foo") InputSuppressWarningsFilter() { }
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;MemberName&quot;/&gt;
+ *   &lt;module name=&quot;ConstantName&quot;/&gt;
+ *   &lt;module name=&quot;ParameterNumber&quot;/&gt;
+ *   &lt;module name=&quot;NoWhitespaceAfter&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private int K; // violation
+ *     &#64;SuppressWarnings({"membername"})
+ *     private int J; // ok
+ *
+ *     private static final int i = 0; // violation
+ *     &#64;SuppressWarnings("checkstyle:constantname")
+ *     private static final int m = 0; // ok
+ *
+ *     public void needsLotsOfParameters (int a, // violation
+ *       int b, int c, int d, int e, int f, int g, int h) {
+ *       // ...
+ *     }
+ *
+ *     &#64;SuppressWarnings("paramnum")
+ *     public void needsLotsOfParameters1 (int a, // ok
+ *       int b, int c, int d, int e, int f, int g, int h) {
+ *       //  ...
+ *     }
+ *
+ *    private int [] ARR; // 2 violations
+ *    &#64;SuppressWarnings("all")
+ *    private int [] ARRAY; // ok
+ * }
  * </pre>
  * <p>
- * Some real check examples:
- * </p>
- * <p>
- * This will prevent from invocation of the MemberNameCheck:
+ * This will prevent selected violations from MemberNameCheck:
  * </p>
  * <pre>
- * &#64;SuppressWarnings({"membername"})
- * private int J;
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;MemberName&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private int K; // violation
+ *     &#64;SuppressWarnings({"membername"})
+ *     private int J; // violation suppressed
+ *
+ * }
  * </pre>
  * <p>
  * You can also use a {@code checkstyle} prefix to prevent compiler from
- * processing these annotations. For example this will prevent ConstantNameCheck:
+ * processing these annotations. For example this will prevent
+ * selected violations from ConstantNameCheck:
  * </p>
  * <pre>
- * &#64;SuppressWarnings("checkstyle:constantname")
- * private static final int m = 0;
+ * &lt;module name=&quot;TreeWalker&quot;&gt;
+ *   &lt;module name=&quot;ConstantName&quot;/&gt;
+ *
+ *   &lt;module name=&quot;SuppressWarningsHolder&quot;/&gt;
+ * &lt;/module&gt;
+ * &lt;module name=&quot;SuppressWarningsFilter&quot;/&gt;
+ * </pre>
+ * <pre>
+ * class Test {
+ *
+ *     private static final int i = 0; // violation
+ *     &#64;SuppressWarnings("checkstyle:constantname")
+ *     private static final int m = 0; // violation suppressed
+ *
+ * }
  * </pre>
  * <p>
  * The general rule is that the argument of the {@code @SuppressWarnings} will be
```

Comment:
From this issue, I am looking for examples that explain the nuances of the code and the property that we were not documenting before.

https://github.com/checkstyle/checkstyle/issues/11641#issuecomment-1126994790
> What documentation may not state, is that it looks like it can also be the Check's ID instead of the name.

We need an example of what it means when check has ID and suppressing by the ID.
https://checkstyle.org/config.html#Id

Have you look at the Java code of the check and tried to determine if there is any other corner cases we are not documenting for `aliasList` or such?

Reply:
@Kevin222004 , please test it.
if it works, better to extend example to let one check have "id" and show how suppressions works by id also.
if it does not work - create issue on this, we can improve later on.

Reply:
I have not find out any Id for membername and constantname here https://github.com/checkstyle/checkstyle/blob/0d9716b2a5dfb42ad1f6249827be67dee7a186f3/src/main/resources/google_checks.xml#L187

do I have tested with ```RightCurly``` https://github.com/checkstyle/checkstyle/blob/0d9716b2a5dfb42ad1f6249827be67dee7a186f3/src/main/resources/google_checks.xml#L93
and it is working 

```
$ cat Test.java
class Test {

  public void test() {

    if (foo) {
      bar();
    } // violation
    else {
      bar();
    }
  }

  @SuppressWarnings("RightCurlyAlone")
  public void test2() {
    if (foo) {
      bar();
    } // violtaion suppressed
    else {
      bar();
    }
  }

}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
	<property name="charset" value="UTF-8" />

<module name="TreeWalker">
  <module name="RightCurly">
    <property name="id" value="RightCurlyAlone"/>
  </module>
   
   <module name="SuppressWarningsHolder">
	</module>
</module>

   <module name="SuppressWarningsFilter" />
</module>

$ java -jar /home/kevin/Downloads/checkstyle-10.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/C_Style/Test.java:7:5: '}' at column 5 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally). [RightCurlyAlone]
Audit done.
Checkstyle ends with 1 errors.
```

Reply:
without adding Property of ``ID`` it won't work I have tested it also

Reply:
Please extend example to put Id to any Check you already have in your examples to show that it is working.
I'd property can be placed to all Checks, and value of id can be any text.